DICOM/src/Microsoft.Health.DeID.SharedLib/EncryptFunction.cs

```diff
@@ -0,0 +1,149 @@
+
+            _aesKey = Encoding.UTF8.GetBytes(encryptionSetting.EncryptKey);
+        }
+
+        public byte[] Encrypt(string plainText, Encoding encoding = null)
+        {
+            EnsureArg.IsNotNull(plainText, nameof(plainText));
+
+            if (plainText == string.Empty)
+            {
+                return new byte[] { };
+            }
+
+            encoding ??= Encoding.UTF8;
+            return Encrypt(encoding.GetBytes(plainText));
+        }
+
+        public byte[] Encrypt(Stream plainStream)
+        {
+            EnsureArg.IsNotNull(plainStream, nameof(plainStream));
+
+            return Encrypt(StreamToByte(plainStream));
+        }
+
+        public byte[] Encrypt(byte[] plainBytes)
+        {
+            EnsureArg.IsNotNull(plainBytes, nameof(plainBytes));
+
+            /* Create AES encryptor:
+             * Mode: CBC
+             * Block size: 16 bytes
+             * Acceptable key sizes： [128, 192, 256]
+             */
+            try
+            {
+                using Aes aes = Aes.Create();
+                byte[] iv = aes.IV;
+                aes.Key = _aesKey;
+                ICryptoTransform encryptor = aes.CreateEncryptor(aes.Key, aes.IV);
+
+                var encryptedBytes = encryptor.TransformFinalBlock(plainBytes, 0, plainBytes.Length);
+
+                // Concat IV to encrpted bytes
+                byte[] result = new byte[AesIvSize + encryptedBytes.Length];
+                Buffer.BlockCopy(iv, 0, result, 0, AesIvSize);
+                Buffer.BlockCopy(encryptedBytes, 0, result, AesIvSize, encryptedBytes.Length);
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                throw new DeIDFunctionException(DeIDFunctionErrorCode.EncryptFailed, "Encrypt data failed.", ex);
+            }
+        }
+
+        public byte[] Decrypt(string cipherText, Encoding encoding = null)
+        {
+            EnsureArg.IsNotNull(cipherText, nameof(cipherText));
+
+            if (cipherText == string.Empty)
+            {
+                return new byte[] { };
+            }
+
+            encoding ??= Encoding.UTF8;
+            var byteData = encoding.GetBytes(cipherText);
+            return Decrypt(byteData);
+        }
+
+        public byte[] Decrypt(Stream cipherStream)
+        {
+            EnsureArg.IsNotNull(cipherStream, nameof(cipherStream));
+
+            return Decrypt(StreamToByte(cipherStream));
+        }
+
+        public byte[] Decrypt(byte[] cipherBytes)
+        {
+            EnsureArg.IsNotNull(cipherBytes, nameof(cipherBytes));
+
+            if (cipherBytes.Length == 0)
+            {
+                return cipherBytes;
+            }
+
+            // Extract IV info from base64 text
+
+            if (cipherBytes.Length < AesIvSize)
+            {
+                throw new FormatException($"The input text for decryption should not be less than {AesIvSize} bytes length!");
```

Comment:
```suggestion
                throw new FormatException($"The length of input data should not be less than {AesIvSize} bytes!");
```

Anyway it's a function for byte[] input 😄